src/unit_tests/syscheckd/test_fim_sync.c

```diff
@@ -119,6 +159,213 @@ static int teardown_json_payload(void **state) {
     return 0;
 }
 
+static int setup_str_pair(void **state) {
+    str_pair_t *new;
+    os_calloc(1, sizeof(str_pair_t), new);
+    os_strdup("first", new->first);
+    os_strdup("last", new->last);
+
+    *state = new;
+    return 0;
+}
+
+
+static int teardown_str_pair(void **state) {
+    str_pair_t *new = *state;
+
+    os_free(new->first);
+    os_free(new->last);
+    os_free(new);
+
+    return 0;
+}
+
+static int setup_fim_entry(void **state) {
+    pair_entry_str_t *data = NULL;
+
+
+    data = malloc(sizeof(pair_entry_str_t));
+    data->entry = malloc(sizeof(fim_entry));
+    data->entry->file_entry.data = malloc(sizeof(fim_file_data));
```

Comment:
You should check that the memory allocation is done correctly. Something like this:
```C
data = malloc(sizeof(pair_entry_str_t));
if (data == NULL) {
    return -1;
}
```